### PR TITLE
[System.Data] fix #22879 (SqlConnection doesn't support empty values)

### LINKED ISSUE #22879: SqlConnection doesn't support empty values
```
var cn = new SqlConnection("Timeout=");
Console.WriteLine(cn.ConnectionTimeout);
```
.NET 4.6 (and mono) prints:
> 15

.NET Core:
> System.ArgumentNullException: 'Value cannot be null.'

I suspect it happens with other parameters too, for example - "packet Size="

### PR DESCRIPTION
**Pull request by EgorBo**
the PR fixes https://github.com/dotnet/corefx/issues/22879 + a small test.

### CODE REVIEW

File: src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs [hunk 1]
```diff
@@ -75,7 +75,7 @@ internal bool ConvertValueToIntegratedSecurityInternal(string stringValue)
         public int ConvertValueToInt32(string keyName, int defaultValue)
         {
             string value;
-            return _parsetable.TryGetValue(keyName, out value) ?
+            return _parsetable.TryGetValue(keyName, out value) && !string.IsNullOrWhiteSpace(value) ?
```
Comment (stephentoub) [inline src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs]: It looks like the reference source is only checking for null, but this is checking for null and whitespace.  Why the extra whitespace check?

Comment (EgorBo) [inline src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs]: @stephentoub Indeed, changed to a simple null-check to match reference sources behavior - tests still passed.

File: src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs [hunk 2]
```diff
@@ -53,5 +53,23 @@ public void SqlConnectionDbProviderFactoryTest()
             Assert.Same(typeof(SqlClientFactory), factory.GetType());
             Assert.Same(SqlClientFactory.Instance, factory);
         }
+
+        [Fact]
+        public void SqlConnectionEmptyParameters()
```
Comment (danmosemsft) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: Seems like 3 separate tests ?

Comment (EgorBo) [inline src/System.Data.SqlClient/tests/FunctionalTests/SqlConnectionTest.cs]: Fixed!

**Comment by corivera** [comment]
Looks like a lot of the other Convert methods were changed in the original commit that caused this issue. Do empty values throw unexpected errors for the non-integer connection string parameters?

**Comment by corivera** [comment]
Looks like the only Convert method I that file that doesn't check for null values is with Integrated Auth: https://github.com/EgorBo/corefx/blob/fbdc9523d227add5da76991d526aef51f6cae566/src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs#L50

**Comment by EgorBo** [comment]
@corivera ah indeed. Fixed!

**Comment by danmosemsft** [comment]
@corivera seems this is good to merge, can we do so?

**Comment by danmosemsft** [comment]
@EgorBo can you resolve conflict?

**Comment by EgorBo** [comment]
@danmosemsft um.. done.
not sure why some build machines failed :(

**Comment by danmosemsft** [comment]
@dotnet-bot test uwp coreclr x64 debug build (https://github.com/dotnet/corefx/issues/23614)

@corivera can you please review, and merge if appropriate?

**Comment by karelz** [comment]
@corivera @saurabh500 @divega ping? Can you please merge it if it is ok?

**Comment by corivera** [comment]
@dotnet-bot Test UWP CoreCLR x64 Debug Build please

**Pull request merged**